modal/_partial_function.py

```diff
@@ -722,3 +730,49 @@ def wrapper(raw_f: Callable[..., Any]) -> _PartialFunction:
         )
 
     return wrapper
+
+
+def _concurrent(
+    _warn_parentheses_missing=None,
+    *,
+    max_inputs: int,  # Hard limit on each container's input concurrency
+    target_inputs: Optional[int] = None,  # Input concurrency that Modal's autoscaler should target
+) -> Callable[[Union[Callable[..., Any], _PartialFunction]], _PartialFunction]:
+    """Decorator that allows individual containers to handle multiple inputs concurrently.
+
+    The concurrency mechanism depends on whether the function is async or not:
+    - Async functions will run inputs on a single thread as asyncio tasks.
+    - Synchronous functions will use multi-threading. The code must be thread-safe.
+
+    Input concurrency will be most useful for workflows that are IO-bound
+    (e.g., making network requests) or when running an inference server that supports
+    dynamic batching.
+
+    When `target_inputs` is set, Modal's autoscaler will try to provision resources such
```

Comment:
Should we mention what happens if target_inputs isn't set but max_inputs is?

Reply:
I.e. that max is then used as the target

Reply:
Also, should a target > max be prevented?

Reply:
You can't not set `max_inputs` when using the decorator, as it doesn't have a default argument.  And we reject `target_inputs > max_inputs` below.

Reply:
I mean if you set the `max` but not the target (that it gets defaulted to the max in that case)

Reply:
Oh I misread. But I'm not following what you think is confusing about the current description. Are you worried people will not realize that modal functions autoscale if `target` is not set?